src/main/scala/org/allenai/plugins/ScaladocGenPlugin.scala

```diff
@@ -0,0 +1,216 @@
+  // linked to, and fixes up the generated HTML to correct scaladoc-style links to javadoc-style
+  // links.
+
+  /** The rt.jar file is located in the path stored in the sun.boot.class.path system property.  See
+    * the Oracle documentation at
+    * http://docs.oracle.com/javase/6/docs/technotes/tools/findingclasses.html.
+    */
+  lazy val rtJar: File = {
+    System.getProperty("sun.boot.class.path").split(JavaFile.pathSeparator).collectFirst {
+      case str: String if str.endsWith(JavaFile.separator + "rt.jar") => str
+    } match {
+      case Some(jar) => file(jar)
+      // Fail hard. We can't reliably generate links.
+      case None => throw new Exception("Error: rt.jar not found; scaladoc generation can't work!")
+    }
+  }
+
+  lazy val scaladocGenJavadocUrl: SettingKey[URL] =
+    settingKey[URL]("The base URL to use for generated Javadoc")
+
+  lazy val scaladocGenExtraJavadocMap: SettingKey[Map[String, URL]] = settingKey[Map[String, URL]](
+    "Mapping for any non-core Java libraries to link in, of library name to Javadoc URL"
+  )
+
+  // Default settings for javadoc generation.
+  /** URL to the Java 8 javadoc. */
+  lazy val javadocUrlSetting =
+    scaladocGenJavadocUrl := url("http://docs.oracle.com/javase/8/docs/api/index.html")
+  /** Extra libraries map, with CoreDependencies java libraries. */
+  lazy val extraJavadocMapSetting = scaladocGenExtraJavadocMap := {
+    Map("slf4j-api" -> url("http://www.slf4j.org/api/"))
+  }
+
+  /** Setting to add API mappings for our Java libraries. This allows the scaladoc generation task
+    * to generate links for these libraries.
+    */
+  lazy val javaApiMappingsSetting = apiMappings ++= {
+    val log = streams.value.log
+    // From the full classpath, find jars with the names given in our doc map.
+    val classpath: Iterable[File] = aggregateFullClasspath.value.flatten map { _.data }
+    val externalJavadocMappings = scaladocGenExtraJavadocMap.value flatMap {
+      case (libraryName, javadocUrl) => {
+        classpath find { file => (file ** s"$libraryName*.jar").get.nonEmpty } map { jar =>
+          jar -> javadocUrl
+        }
+      }
+    }
+    // Add the runtime mapping to the mappings we just generated & return.
+    externalJavadocMappings + (rtJar -> scaladocGenJavadocUrl.value)
+  }
+
+  /** Setting that overrides the doc generation task. This finds all HTML files generated by the doc
+    * task, then subs in the correctly-formatted Javadoc link in place of the badly-formatted
+    * Scaladoc link.
+    */
+  lazy val fixJavadocSetting = doc in ScalaUnidoc := {
+    // Build up a regular expression that matches any of the external javadoc links we have.
+    val externalUrls =
+      scaladocGenJavadocUrl.value +: scaladocGenExtraJavadocMap.value.values.toVector
+    val anyUrl = externalUrls map { url => s"""\\Q$url\\E""" } mkString { "|" }
+    val captureUrlAndFragment = s""""($anyUrl)#([^"]*)"""".r
```

Comment:
could you add a comment showing an example fully-baked regex? Would be easier to parse what's going on here.